### PR TITLE
Update design note and class diagrams with up to date information

### PR DESCRIPTION
**Pull request by AdrianMontaguSmartDCSIT**
This PR updates design note 3 and class diagrams with up to date models.

### CODE REVIEW

File: docs/development/design/3-DN-Retrieving_unique_graphs_from_Otel_data/class_diagrams.puml [hunk 1]
```diff
@@ -8,35 +8,30 @@ struct OTelEvent {
...
-    #data_source: str
-    +__init__(): None
     +__iter__(): Self
     +__next__(): OTelEvent
-    +get_yaml_config(): Any
-    +set_file_ext(): str
 }
 
 class JSONDataSource {
+    #config: JSONDataSourceConfig
```
Comment (FreddieMatherSmartDCSIT) [inline docs/development/design/3-DN-Retrieving_unique_graphs_from_Otel_data/class_diagrams.puml:21]: Should we maybe add to the class diagram this class and its dependencies as we are mentioning it?

**Review by FreddieMatherSmartDCSIT — CHANGES_REQUESTED** [review]
Just thinking that we can add the JSONDataSourceConfig here and its dependencies

File: docs/development/design/3-DN-Retrieving_unique_graphs_from_Otel_data/class_diagrams.puml (hunk 1 above)
Comment (AdrianMontaguSmartDCSIT) [reply docs/development/design/3-DN-Retrieving_unique_graphs_from_Otel_data/class_diagrams.puml:21]: Yes good point

**Review by FreddieMatherSmartDCSIT — APPROVED** [review]
Looks good to me

**Pull request merged**